### PR TITLE
MM-16844 "Fixing"

### PR DESCRIPTION
**Pull request by crspeller**
Refactor and adding test framework. Fixing a couple issues so it works at all.

@jasonblais However I think we should cut it from our list of plugins to pre-package as it mostly doesn't work as intended. 
Example:
Let's meet today at 10am ET
would render as:
Let's meet `today at 10am ET` *(Wed, Jul 17 8:00 AM PDT)*

which is incorrect. The correct time would be 7am. The reason for this issue is the plugin being confused over timezone abbreviations and daylight savings. I spent a bit of time trying to find a solution to this, but current javascript libraries don't allow for the needed adjustments.

### CODE REVIEW

**Comment by mkraft** [comment]
@crspeller "ET" is not a supported time zone abbreviation. EDT or EST.

**Comment by mkraft** [comment]
@jasonblais @crspeller There is a big usability issue with this plugin: having to include the time zone abbreviation. Problems include:

* Remembering to include (and knowing) the correct time zone abbreviation initialism (ex EST)
* Remembering to use (and knowing) the daylight saving time variant based on the date (ex EDT)

This requirement can be eliminated by storing the walltime of the post author at the time of the post creation. This could be achieved by either including this change in the core product or perhaps as a part of the data stored on the post by the plugin itself.

File: webapp/src/test.js [hunk 1]
```diff
@@ -0,0 +1,17 @@
+import {convertTimesToLocal} from './time';
+
+test('convertTimesToLocal', () => {
+    const testCases = [
+
+        // This is an incorrect test case that demonstrates the issue with this plugin. 10am ET is not 8am pacific time at the refrence time.
+        // Additional tests shoudl be written if we find a solution to this.
+        {
+            test: "Let's meet today at 10am ET",
```
Comment (cpoile) [inline webapp/src/test.js]: try one with EDT?

Comment (crspeller) [reply webapp/src/test.js]: Ya it works if you specify the correct daylight savings zone. But nobody's going to do that :P

**Pull request merged**